src/SFML/Window/Win32/JoystickImpl.cpp

```diff
@@ -861,6 +1081,100 @@ JoystickCaps JoystickImpl::getCapabilitiesDInput() const
     return caps;
 }
 
+////////////////////////////////////////////////////////////
+JoystickState JoystickImpl::updateXInput()
+{
+    // No identified index yet
+    if (m_xInputIndex == 0xFFFFFFFF)
+    {
+        for (std::size_t i = 0; i < xInputSlots.size(); ++i)
+        {
+            const bool taken = xInputSlots[i];
+            if (!taken)
+            {
+                xInputSlots[i] = true;
+                m_xInputIndex  = static_cast<DWORD>(i);
+                break;
+            }
+        }
+    }
+
+    XINPUT_STATE xinputState{};
+    const auto   result = mXInputGetState(m_xInputIndex, &xinputState);
```

Comment:
You seem to be choosing your own index here, but it is actually XInput that assigns this ID.
You can't just query on a random ID and expect that to be the actual controller you are working with.

Reply:
Yes, it seems that I was naive and tried to preempt which slot a newly added device would occupy. 

I have pushed further changes, please test them when you have the time.